test/lit/passes/optimize-instructions-sign-ext.wast

```diff
@@ -62,4 +62,34 @@
     (drop (i64.shr_s (i64.shl (local.get $x) (i64.const 16)) (i64.const 16))) ;; skip
     (drop (i64.extend_i32_s (i32.wrap_i64 (local.get $x))))
   )
+
+  ;; i32(x) << 24 >> 24   ==>   i32.extend8_s(x)
+  ;; i32(x) << 16 >> 16   ==>   i32.extend16_s(x)
+
+  ;; CHECK:      (func $i32-sign-extentions (param $x i32)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.extend8_s
+  ;; CHECK-NEXT:    (local.get $x)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.extend16_s
+  ;; CHECK-NEXT:    (local.get $x)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.shr_s
+  ;; CHECK-NEXT:    (i32.shl
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:     (i32.const 8)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (i32.const 8)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $i32-sign-extentions (param $x i32)
+    (drop (i32.shr_s (i32.shl (local.get $x) (i32.const 24)) (i32.const 24)))
+    (drop (i32.shr_s (i32.shl (local.get $x) (i32.const 16)) (i32.const 16)))
+    (drop (i32.shr_s (i32.shl (local.get $x) (i32.const 8)) (i32.const 8))) ;; skip
+  )
```

Comment:
Please add a test where the constants do not match. Say 24 and 16 (so each is valid by itself).

Reply:
Also a test where everything is good to optimize except for the outer shift being unsigned.

Reply:
done!